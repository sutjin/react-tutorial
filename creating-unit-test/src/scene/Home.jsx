import React, { Component } from "react";

class Home extends Component {
  constructor(props) {
    super(props);

    this._increaseCount = this._increaseCount.bind(this);

    this.state = {
      count: 0
    };
  }

  _increaseCount() {
    const increase = this.state.count + 1;
    this.setState({count: increase});
  }

  render() {
    return (
      <div id="main_wrapper">
        <h1>Hello World</h1>
        {this.props.text &&
          <p>{this.props.text}</p>
        }
        {this.props.function &&
          <div className="test_enter" onMouseEnter={this.props.function} />
        }
        <p>Current Count {this.state.count}</p>
        <button type="button" onClick={this._increaseCount}>Click Me!</button>
      </div>
    );
  }
}

export default Home;
