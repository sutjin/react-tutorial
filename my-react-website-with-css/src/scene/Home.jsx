import React, { Component } from "react";
import ReactDOM from "react-dom";
import Button from "../component/Button.jsx";

import image from "../media/react_logo.png";

class Home extends Component {
  render() {
    return (
      <div id="main_wrapper">
        <h1>Hello Wold</h1>
        <Button />
        <img src={image} />
      </div>
    );
  }
}

export default Home;

// responsible in finding the DOM that we will render to
const wrapper = document.getElementById("react_container");
wrapper ? ReactDOM.render(<Home />, wrapper) : false;
