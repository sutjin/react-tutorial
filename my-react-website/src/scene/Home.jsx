import React, { Component } from "react";
import ReactDOM from "react-dom";
// importign the button class that we built previously
import Button from "../component/Button.jsx";

class Home extends Component {
  render() {
    return (
      <div id="main_wrapper">
        <h1>Hello Wold</h1>
        <Button />
      </div>
    );
  }
}

export default Home;

// responsible in finding the DOM that we will render to
const wrapper = document.getElementById("react_container");
wrapper ? ReactDOM.render(<Home />, wrapper) : false;
