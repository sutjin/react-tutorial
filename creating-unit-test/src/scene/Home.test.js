import React from 'react';
import Home from './Home.jsx';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('<Home />', () => {
  it('renders <Home /> container correctly', () => {
    const tree = renderer
      .create(
        <Home />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Home with <p> for content', () => {
    const tree = renderer
      .create(
        <Home
          text="Nice to meet you"
        />
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should increase the state count by one', () => {
    const wrapper = shallow(<Home />);
    // verify initial state
    expect(wrapper.state("count")).toEqual(0);
    // "click" it once and see if value increase
    wrapper.find('button').simulate('click');
    expect(wrapper.state("count")).toEqual(1);
  });

  it("should trigger hover function on hover", () => {
    const hoverFunction = jest.fn(), // Mock function that we can trigger
      wrapper = shallow(<Home function={hoverFunction}/>);

    wrapper.find('div.test_enter').simulate('mouseenter');
    // verify that the mocked function has been triggered
    expect(hoverFunction).toHaveBeenCalled();
  });
});
