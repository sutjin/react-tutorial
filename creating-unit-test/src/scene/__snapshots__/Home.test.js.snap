// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders <Home /> container correctly 1`] = `
<div
  id="main_wrapper"
>
  <h1>
    Hello World
  </h1>
  <p>
    Current Count 
    0
  </p>
  <button
    onClick={[Function]}
    type="button"
  >
    Click Me!
  </button>
</div>
`;

exports[`<Home /> renders Home with <p> for content 1`] = `
<div
  id="main_wrapper"
>
  <h1>
    Hello World
  </h1>
  <p>
    Nice to meet you
  </p>
  <p>
    Current Count 
    0
  </p>
  <button
    onClick={[Function]}
    type="button"
  >
    Click Me!
  </button>
</div>
`;
